### ISSUE TITLE
TypeError: Cannot assign to read only property 'length' of Error

### ISSUE THREAD

**Issue by binarykitchen** [petkaantonov/bluebird#291]
```
    UncaughtException: TypeError: Cannot assign to read only property 'length' of Error
        at Object.<anonymous> (/home/michael.heuberger/binarykitchen/code/videomail.io/node_modules/sharp/node_modules/bluebird/js/main/errors.js:84:33)
        at Module._compile (module.js:456:26)
        at Object.Module._extensions..js (module.js:474:10)
        at Module.load (module.js:356:32)
        at Function.Module._load (module.js:312:12)
        at Module.require (module.js:364:17)
        at require (module.js:380:17)
        at module.exports (/home/michael.heuberger/binarykitchen/code/videomail.io/node_modules/sharp/node_modules/bluebird/js/main/promise.js:34:14)
        at Object.<anonymous> (/home/michael.heuberger/binarykitchen/code/videomail.io/node_modules/sharp/node_modules/bluebird/js/main/bluebird.js:24:38)
        at Module._compile (module.js:456:26)
```

Using https://github.com/lovell/sharp here and unable to open an image due to the above error.

It's this code line

```
AggregateError.prototype.length = 0;
```

I think you cannot do that ...

**Comment by bf** [comment]
I experience this error message with bluebird and redis-lock after trying to install my app from scratch. It's crazy that some module so far down the dependency line is able to crash everything..

**Comment by petkaantonov** [comment]
@bf 

Can you manually edit the line to:

```js
Object.definePropertty(AggregateError.prototype, "length", {
      value: 0,
      configurable: false,
      writable: true,
      enumerable: true
});
```

And see if that fixes it?

**Comment by petkaantonov** [comment]
@bf should be fixed in 2.9.9

**Comment by bf** [comment]
@petkaantonov thanks, that fixed it! That was by far the most responsive npm issue I saw resolved! Thanks again!
